contracts/OrderMixin.sol

```diff
@@ -114,14 +103,20 @@ abstract contract OrderMixin is IOrderMixin, EIP712, PredicateHelper {
     /**
      * @notice See {IOrderMixin-cancelOrder}.
      */
-    function cancelOrder(OrderLib.Order calldata order) external returns(uint256 orderRemaining, bytes32 orderHash) {
-        if (order.maker != msg.sender) revert AccessDenied();
-
-        orderHash = hashOrder(order);
-        orderRemaining = _remaining[orderHash];
+    function cancelOrder(bytes32 orderHash) public returns(uint256 orderRemaining) {
+        orderRemaining = _remaining[msg.sender][orderHash];
         if (orderRemaining == _ORDER_FILLED) revert AlreadyFilled();
         emit OrderCanceled(msg.sender, orderHash, orderRemaining);
```

Comment:
Why emit OrderCanceled is before last assignment?

Reply:
Usually it makes no difference where exactly in method event emits, usually we try to locate it in the most comfortable place to have all necessary arguments in local vars. But in this method it seems like it was legacy location, now I don’t see any reason behind the location not in the latest line.